com.unity.perception/Runtime/Randomization/Parameters/CategoricalParameter.cs

```diff
@@ -175,9 +175,17 @@ void NormalizeProbabilities()
         public T Sample()
         {
             var randomValue = m_Sampler.Sample();
-            return uniform
-                ? m_Categories[(int)(randomValue * m_Categories.Count)]
-                : m_Categories[BinarySearch(randomValue)];
+            if (uniform)
+            {
+                var index = (int)(randomValue * m_Categories.Count);
+                if (index == m_Categories.Count)
```

Comment:
Do you think we could inline this operation?

`index = index == m_Categories.Count ? index - 1 : index;`

Just a personal preference thing. Feel free to ignore this